openedx/core/djangoapps/theming/helpers.py

```diff
@@ -17,7 +23,30 @@ def get_value(val_name, default=None, **kwargs):
     """
     This is a proxy function to hide microsite_configuration behind comprehensive theming.
     """
-    return microsite.get_value(val_name, default=default, **kwargs)
+    # Attempt to retrieve the requested field/value from the microsite configuration
+    microsite_value = microsite.get_value(val_name, default=default, **kwargs)
+
+    # If we received a non-dictionary value from the microsite config workflow, return it immediately.
+    if not isinstance(microsite_value, dict):
+        return microsite_value
+
+    # Load the default configuration value so we can take a closer look
+    if not default:
+        default_dict = getattr(settings, val_name)
+    else:
+        default_dict = default
+
+    # If the default value is not a dictionary, something fishy is going on...
+    if not isinstance(default_dict, dict):
+        msg = "Type for microsite setting '{name}' does not match the base setting type.".format(name=val_name)
+        log.warn(msg)
+        return microsite_value
+
+    # We have two dicts at this point, so override some/all of the default dictionary
+    # with the microsite configuration dictionary's fields+values
+    override_dict = dict(default_dict)
+    override_dict.update(microsite_value)
+    return override_dict
```

Comment:
Do we want to lose the settings lookup using `val_name` if no default setting is provided?